modin/pandas/test/test_io.py

```diff
@@ -751,19 +751,38 @@ def test_to_pickle():
 
 
 def test_to_sql():
```

Comment:
Let's try to keep the tests smaller and properly utilize the `pytest.fixtures` by
1. Splitting up `test_to_sql` into two tests, `test_to_sql_without_index` and `test_to_sql_with_index` that take in a `pytest.fixture`
2. Merging the `pytest.fixture` `teardown_sql_file` and `setup_sql_file` into a new `pytest.fixture` called, possibly, `sql_connection` that creates and tears down the sql connection ([source](https://docs.pytest.org/en/latest/fixture.html#fixture-finalization-executing-teardown-code))